packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -59,50 +74,86 @@ export const SampleFinderHeaderButtons: FC<SampleFinderHeaderProps> = memo(props
     );
 });
 
+
+function getLocalStorageKey(): string {
+    return getPrimaryAppProperties().productId + ActionURL.getContainer() + "-SampleFinder";
+}
+
 export const SampleFinderSection: FC<Props> = memo(props => {
     const { parentEntityDataTypes, ...gridProps } = props;
 
+    const [filterChangeCounter, setFilterChangeCounter] = useState<number>(0);
     const [chosenEntityType, setChosenEntityType] = useState<EntityDataType>(undefined);
-    const [filterCards, setFilterCards] = useState<FilterCardProps[]>([]);
+    const [filters, setFilters] = useState<FilterProps[]>([]);
+
+    useEffect(() => {
+        const finderSessionDataStr = sessionStorage.getItem(getLocalStorageKey());
+        if (finderSessionDataStr) {
+            const finderSessionData = JSON.parse(finderSessionDataStr);
+            if (finderSessionData.filters) {
+                setFilters(finderSessionData.filters);
+            }
+            if (finderSessionData.filterChangeCounter !== undefined) {
+                setFilterChangeCounter(finderSessionData.filterChangeCounter);
+            }
+        }
+    }, []);
+
+    const getSelectionKeyPrefix = (): string => {
+        return 'sampleFinder-' + filterChangeCounter;
+    };
+
+    const updateSessionStorage = (filterChangeCounter: number, filters: FilterProps[]) => {
+        sessionStorage.setItem(getLocalStorageKey(), JSON.stringify({
+            filterChangeCounter,
+            filters
+        }));
+    };
 
     const onAddEntity = useCallback((entityType: EntityDataType) => {
         setChosenEntityType(entityType);
     }, []);
 
     const onFilterEdit = useCallback(
         (index: number) => {
-            console.log('onFilterEdit for index ' + index + ': Not yet implemented.');
+            setFilterChangeCounter(filterChangeCounter+1);
             setChosenEntityType(parentEntityDataTypes[index]);
+            // TODO update filters as well
+            updateSessionStorage(filterChangeCounter + 1, filters);
         },
-        [parentEntityDataTypes]
+        [parentEntityDataTypes, filterChangeCounter]
     );
 
     const onFilterDelete = useCallback(
         (index: number) => {
-            const newFilterCards = [...filterCards];
+            const newFilterCards = [...filters];
             newFilterCards.splice(index, 1);
-            setFilterCards(newFilterCards);
+            setFilters(newFilterCards);
+            setFilterChangeCounter(filterChangeCounter+1);
+            updateSessionStorage(filterChangeCounter + 1, newFilterCards);
         },
-        [filterCards]
+        [filters, filterChangeCounter]
     );
 
     const onFilterClose = () => {
         setChosenEntityType(undefined);
+        setFilterChangeCounter(filterChangeCounter+1);
```

Comment:
it looks like the only usage of this function below also calls `setFilterChangeCounter` with a + 1, so this is likely updating that counter twice.